src/main/java/seedu/address/model/person/Priority.java

```diff
@@ -0,0 +1,89 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Represents a Person's priority in the address book.
+ * The priority can only be one of these values: "LOW", "MEDIUM", or "HIGH".
+ *
+ * Guarantees: The priority is immutable and always valid.
+ */
+public class Priority {
+
+    // Regex to validate the priority value (LOW, MEDIUM, or HIGH).
+    public static final String VALIDATION_REGEX = "LOW|MEDIUM|HIGH";
+
+    // Error message for invalid priority values.
+    public static final String MESSAGE_CONSTRAINTS = "Priority should only be 'LOW', 'MEDIUM', or 'HIGH'.";
+
+    private final String priority;
+
+    /**
+     * Constructs a {@code Priority} object with the specified priority string.
+     *
+     * @param priority The priority value for the person.
+     *                 Must be either "LOW", "MEDIUM", or "HIGH".
+     * @throws IllegalArgumentException if the {@code priority} is invalid.
+     */
+    public Priority(String priority) {
+        requireNonNull(priority);
+        if (!isValidPriority(priority)) {
+            throw new IllegalArgumentException(MESSAGE_CONSTRAINTS);
```

Comment:
Could use the existing `checkArgument` method so that this can all be done in 1 line of code and to maintain SLAP

Reply:
Done. Thanks for catching that and pointing that out! Overlooked on my part.